### ISSUE TITLE
How can we add custom namespace declarations to gpx element?

### ISSUE THREAD

**Issue by lorenh** [NetTopologySuite/NetTopologySuite.IO.GPX#36]
Is injecting a custom namespace similar to the following possible when writing a GPX with this library?

```
<gpx version ="1.1"
          xmlns="http://www.topografix.com/GPX/1/1"
          xmlns:ABC="http://www.mycustomnamespace.net/xml">
:
</gpx>
```

**Comment by airbreather** [comment]
Not at the top level; the only place where you have an opportunity to inject custom stuff like that is when we go to write out the children of `<extensions>` elements.

Essentially, you'd make a subclass of [`GpxExtensionWriter`](https://nettopologysuite.github.io/NetTopologySuite.IO.GPX/api/NetTopologySuite.IO.GpxExtensionWriter.html) and override what you need.  The namespaces of your `XElement` instances should get written out correctly if necessary, though I think it will be redundant, and I'm not 100% sure off the top of my head if it's going to pick the namespace prefix you want it too.

**Comment by lorenh** [comment]
OK, that's what I thought. Technically speaking the XML namespace declaration should be interpreted by the XML parser the same whether it's at the top or duplicated on each element.  I was just trying to match a spec given to me. Now I know I can't do it quite the way they want. Hopefully this will be sufficient.  But keep it in mind if you are ever in there tinkering with this area in case some way to customize things like this pops into your head.

**Comment by airbreather** [comment]
I wouldn't mind adding the option to accept a list of namespace / prefix pairs in `GpxWriterSettings`, if for no other reason than to shrink the output for these cases.

**Comment by lorenh** [comment]
OK, I guess I'd like that feature added when you get around to it.

**Comment by lorenh** [comment]
I created a pull request with one way you could achieve this, I'm sure you can improve on it.

https://github.com/NetTopologySuite/NetTopologySuite.IO.GPX/pull/37

**Comment by lorenh** [comment]
Still curious if this is on a roadmap for a future release?  I'd love to abandon using the custom package I built.

**Comment by airbreather** [comment]
Yes -- I'm focusing most of my free time on the NTS v2 upgrades, so this is going to be a while.

**Comment by lorenh** [comment]
I updated the PR for this to merge cleanly into 1.0 branch in case you ever get around to it.  Again, I'd love to abandon using my custom package.

**Comment by airbreather** [comment]
Resolved by #37, but I'm going to see if I can make it any better before releasing a 1.1.0 that includes it.

**Comment by airbreather** [comment]
1.1.0 is going to be on NuGet with the most important fix for this.  I'll open a new issue to track parsing out existing namespaces on read for the sake of a better round-trip.
